include/deal.II/base/index_set.h

```diff
@@ -229,6 +229,12 @@ class IndexSet
   void compress () const;
 
   /**
+   * Reorder the ranges inside IndexSet so that the largest range is the first
+   * one.
+   */
+  void reorder_ranges();
```

Comment:
It should be fine. I still need to modify ```index_within_set``` because the binary search is slightly different but I don't think that there will be any problem for member functions. We may have problems when we start using this class somewhere else in the library but I don't think that using a range + an IndexSet would simplify the problem.